commit c601098b6857691e9b769dad7bc87ba54b29b136 (refs/changes/44/2144/3)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-06-23T22:07:40+00:00 (2 years, 4 months ago)
    
    DOCKER-1072 Work around OS-6193 in tools/check-docs.sh
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Approved by: Trent Mick <trent.mick@joyent.com>

diff --git a/deps/javascriptlint b/deps/javascriptlint
index e1bd0ab..ad52812 160000
--- a/deps/javascriptlint
+++ b/deps/javascriptlint
@@ -1 +1 @@
-Subproject commit e1bd0abfd424811af469d1ece3af131d95443924
+Subproject commit ad52812e77bdfb1e90fb71a1201adb2b665a27e6
diff --git a/tools/check-docs.sh b/tools/check-docs.sh
index cbba338..1507777 100755
--- a/tools/check-docs.sh
+++ b/tools/check-docs.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -36,10 +36,9 @@ TOP=$(cd $(dirname $0)/../; pwd)
 
 hits=0
 for file in $(find $TOP/docs -name "*.md"); do
-    lastchar=$(tail -c1 $file | od -a | head -1 | awk '{print $2}')
-    if [[ $lastchar != 'nl' ]]; then
+    if [[ "`tail -1c $file`" != "`printf '\n'`" ]]; then
         echo "$file: does not end with a newline" >&2
-        hits=$(( $hits + 1 ))
+        hits=1
     fi
 done
 
